data-processor/src/test/groovy/io/micronaut/data/processor/visitors/FindSpec.groovy

```diff
@@ -249,7 +249,7 @@ interface PlayerRepository extends GenericRepository<Player, Integer> {
 
         expect:
         findByTeamName.stringValue(Query).get() == 'SELECT player_ FROM test.Player AS player_ JOIN player_.team player_team_ WHERE (player_team_.name = :p1)'
-        findByTeamId.stringValue(Query).get() == 'SELECT player_ FROM test.Player AS player_ WHERE (player_.team.id = :p1)'
+        findByTeamId.stringValue(Query).get() == 'SELECT player_ FROM test.Player AS player_ JOIN player_.team player_team_ WHERE (player_.team.id = :p1)'
```

Comment:
This to me suggests the fix is valid, otherwise this query might not get executed since association is not joined.

Reply:
Hm, this is not correct since it adds join but searches using `WHERE (player_.team.id = :p1)` instead of `WHERE (player_team_.id = :p1)`